packages/astro/src/core/add/index.ts

```diff
@@ -407,7 +407,7 @@ async function getInstallIntegrationsCommand({
 		.map<[string, string | null][]>((i) => [[i.packageName, null], ...i.dependencies])
 		.flat(1)
 		.filter((dep, i, arr) => arr.findIndex((d) => d[0] === dep[0]) === i)
-		.map(([name, version]) => (version === null ? name : `${name}@${version}`))
+		.map(([name, version]) => (version === null ? name : `${name}@${version.split(' ').pop()}`))
```

Comment:
Would this end up always installing `react@^17`?

I'd almost expect `astro add` to prompt me for which version of react I want, 

```sh
**Which version of react would you like to install?**
◯  ^17.0.2
◯  ^18.0.0
```

Reply:
Agreed! I still think we can get this in and add that as a future enhancement too 👍